### DISCUSSION TITLE
RAX3000M 刷固件时没刷BL2，还需要重新刷入吗？

### DISCUSSION THREAD

**Discussion by pop2006** [immortalwrt/immortalwrt#1152] [Q&A]
### Describe the bug

刷系统时没有看到mediatek: add CMCC RAX3000M support #1075文档。想当然的刷入uboot后重启，发现无法登录网页，经别人提醒才知道要要用ftfpd来传固件。
现在系统是正常的，但我想重新刷写BL2时，已经提示不能写入了。
在网上搜索了一下，发现有人写了个H3C Magic NX30 Pro 官方 OpenWrt 安装教程，里面也有个刷完系统后再刷BL2的，照着做，但是也一样无法刷写。https://ericclose.github.io/install-openwrt-on-h3c_magic-nx30-pro.html
请教应该怎么办？

### ImmortalWrt version

r27304-31bc47589e

### ImmortalWrt release

23.05.1

### ImmortalWrt target/subtarget

mediatek/filogic

### Device

CMCC RAX3000M

### Image kind

Official downloaded image

### Terms

- [ ] I am reporting an issue for ImmortalWrt, not an unsupported fork.

**Comment by 0x5826** [comment]
给你总结一下吧
1. 如果你原来有带web的uboot，可以不刷BL2，然后使用**支持NMBM**的固件是没有问题的。
2. 如果你要刷BL2那么对应的FIP也要刷，那么此时uboot是**不带NMBM**支持的，你必须通过ttfp刷写**不带NMBM**的固件。
也就是说目前两种uboot支持两种固件，一种是237、hanwckf仓库的固件都是开启了NMBM支持的（imm仓库也有支持NMBM），在uboot web页面刷入即可；另一种是官方op和imm仓库中不带NMBM支持的，需要刷FIP和BL2，而且需要tftp方式刷入。

**Reply by debsir** [reply]
请问：我也是刷的官方op，没刷BL2会有什么后果呢？跟NMBM有关吗？此外，我看到的教程FIP和BL2都是用mtd刷入的，可以用tftp来刷吗？
